filament/backend/src/opengl/OpenGLProgram.cpp

```diff
@@ -203,6 +210,29 @@ void OpenGLProgram::initializeProgramState(OpenGLContext& context, GLuint progra
         }
     }
     mUsedBindingsCount = usedBindingCount;
+
+    // Push constant initialization
+    auto& constants = lazyInitializationData.pushConstants;
+    if (!constants.empty()) {
+        mVertexPushConstants.reserve(constants.size());
+        mVertexPushConstants.resize(constants.size());
+        // The constants are defined in a struct of name PUSH_CONSTANT_STRUCT_VAR_NAME. We prepend
+        // the prefix here to avoid string manipulation in a the draw loop.
+        std::stringbuf constantPrefix;
+        constantPrefix.sputn(Program::PUSH_CONSTANT_STRUCT_VAR_NAME,
+                strlen(Program::PUSH_CONSTANT_STRUCT_VAR_NAME));
+        constantPrefix.sputn(".", 1);
```

Comment:
you can probably implement this with just a `snprintf()`. Try to avoid the STL for string manipulation as it brings it a lot of code.

Reply:
done